src/test/java/com/teragrep/pth_06/walker/ConditionWalkerTest.java

```
@@ -326,14 +320,36 @@ void multipleSearchTermTwoAndOneMatchTest() {
+                + "      and \"bloomdb\".\"pattern_test_ip\".\"filter\" is not null\n" + "    )\n" + "    or (\n"
+                + "      bloommatch(\n" + "        (\n" + "          select \"term_0_pattern_test_ip255\".\"filter\"\n"
+                + "          from \"term_0_pattern_test_ip255\"\n" + "          where (\n" + "            term_id = 0\n"
+                + "            and type_id = \"bloomdb\".\"pattern_test_ip255\".\"filter_type_id\"\n" + "          )\n"
+                + "        ),\n" + "        \"bloomdb\".\"pattern_test_ip255\".\"filter\"\n" + "      ) = true\n"
+                + "      and \"bloomdb\".\"pattern_test_ip255\".\"filter\" is not null\n" + "    )\n" + "    or (\n"
+                + "      \"bloomdb\".\"pattern_test_ip\".\"filter\" is null\n"
+                + "      and \"bloomdb\".\"pattern_test_ip255\".\"filter\" is null\n" + "    )\n" + "  )\n"
+                + "  and (\n" + "    (\n" + "      bloommatch(\n" + "        (\n"
+                + "          select \"term_1_pattern_test_ip\".\"filter\"\n"
+                + "          from \"term_1_pattern_test_ip\"\n" + "          where (\n" + "            term_id = 1\n"
+                + "            and type_id = \"bloomdb\".\"pattern_test_ip\".\"filter_type_id\"\n" + "          )\n"
+                + "        ),\n" + "        \"bloomdb\".\"pattern_test_ip\".\"filter\"\n" + "      ) = true\n"
+                + "      and \"bloomdb\".\"pattern_test_ip\".\"filter\" is not null\n" + "    )\n"
+                + "    or \"bloomdb\".\"pattern_test_ip\".\"filter\" is null\n" + "  )\n" + ")";
         Condition cond = Assertions.assertDoesNotThrow(() -> walker.fromString(q, false));
         Assertions.assertEquals(e, cond.toString());
+        // expect 2 tables here because we allow for other conditions after the without filters condition that can add tables to the required tables
```

i think it would be better to throw an exception that if indexStatement (search word) is present and withoutFilters is set. otherwise the behaviour is a bit mysterious to the user. i don't find use case for the combined.

---

Sure I was on the fence about it also. I will refactor so that it throws an exception in this case

---

Refactored and added test for exception